numpyro/distributions/directional.py

```diff
@@ -14,14 +14,19 @@
 
 @copy_docs_from(Distribution)
 class VonMises(Distribution):
-    arg_constraints = {'loc': constraints.interval(-math.pi, math.pi), 'concentration': constraints.positive}
+    arg_constraints = {'loc': constraints.real, 'concentration': constraints.positive}
 
-    @property
-    def support(self):
-        return constraints.interval(-math.pi, math.pi)
+    support = constraints.interval(-math.pi, math.pi)
 
     def __init__(self, loc, concentration, validate_args=None):
-        self.loc, self.concentration = promote_shapes(loc, concentration)
+        """  von Mises distribution for sampling directions.
+
+        :param loc: center or distribution
+                    NOTE: mapped to [-pi, pi]
+        :param concentration: concentration of distribution
+        """
+        # Map loc to [-pi, pi]
+        self.loc, self.concentration = promote_shapes((loc + jnp.pi) % (2. * jnp.pi) - jnp.pi, concentration)
```

Comment:
I think you don't need to map `loc` to `(-pi, pi)` (so we can save some unnecessary computations). If users want to access the mean of VonMises `d`, they can do `d.mean`. (another minor benefit of keeping `loc` as-is is: if `loc` is the result of a regression model, then we can easily access the original location through `d.loc` of the model trace) Both `sample` and `log_prob` methods already implicitly do this "modulo 2pi" job. Also, keeping the original value agrees with its constraint `real`. WDYT?

Reply:
I agree with your last point; keeping the original value means the domain is consistent. I added that "mean is the same as loc mapped to support" to docstring for `mean.`